netlify.toml

```diff
@@ -1,12 +1,9 @@
 [build.environment]
   NODE_VERSION = "18"
+  NODE_OPTIONS = "--max_old_space_size=4096"
 
 [build]
   publish = "docs/.vitepress/dist"
   command = "pnpm run docs:build"
 
-[[redirects]]
-  from = "/*"
-  to = "/index.html"
-  status = 200
```

Comment:
I would prefer VitePress to fix this instead of workaround here.

Reply:
@antfu VP cannot fix this since Netlify is responding with 200 forwarding index.html content (the missing page is there), you should remove SPA from Netlify, then Netlify will send 404 and VP will show 404 page. 

If you mean VP detects that the route doesn't exist and redirects to 404, I'll take a look at that later.

FYI: I've removed SPA from Vite PWA and Vitest docs (and also from my vuetify nuxt module).

Reply:
What do you mean by "remove SPA"? Removing this redirection would just work?

Reply:
yes, check this url https://deploy-preview-16--eslint-stylistic.netlify.app/guide/missing and then navigate, check the same url in production